inference-engine/thirdparty/movidius/XLink/shared/include/XLinkSemaphore.h

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///
+/// @file
+///
+/// @brief     Application configuration Leon header
+///
+
+#ifndef _XLINKSEMAPHORE_H
+#define _XLINKSEMAPHORE_H
+
+# if (defined(_WIN32) || defined(_WIN64))
+#  include "win_pthread.h"
+#  include "win_semaphore.h"
+#  include "win_synchapi.h"
+# else
+#  include <pthread.h>
+#  ifdef __APPLE__
+#   include "pthread_semaphore.h"
+#  else
+#   include <semaphore.h>
+# endif
+# endif
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+//
+// This structure describes the semaphore used in XLink and
+// extends the standard semaphore with a reference count.
+// The counter is thread-safe and changes only in cases if
+// all tools of thread synchronization are really unlocked.
+// refs == -1 in case if semaphore was destroyed;
+// refs == 0 in case if semaphore was initialized but has no waiters;
+// refs == N in case if there are N waiters which called sem_wait().
+//
+
+typedef struct {
+    sem_t psem;
+    int refs;
```

Comment:
What version of C is used? Is it C11? If so [atomic](https://en.cppreference.com/w/c/language/atomic) types could be used instead of `int` + mutex